src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -184,10 +205,9 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
                 ArrayBuilder<DiagnosticData> list,
                 CancellationToken cancellationToken)
             {
+                analyzers = analyzers.WhereAsArray(a => MatchesPriority(a));
                 if (analyzers.IsEmpty)
-                {
                     return;
-                }
 
                 var analysisScope = new DocumentAnalysisScope(_document, span, analyzers, kind);
```

Comment:
Instead of filtering above, pass in the priority here and let `DocumentAnalysisExecutor` do the filtering. Basically, `executor.ComputeDiagnosticsAsync` will have an early bail out check for priority matching.